packages/plugin-spatial-nback/src/index.ts

```diff
@@ -0,0 +1,509 @@
+      },
+      /** Color of incorrect feedback border */
+      incorrect_color: {
+        type: ParameterType.STRING,
+        default: "#cc0000",
+      },
+      /** Instructions to display above the grid */
+      instructions: {
+        type: ParameterType.STRING,
+        default: "Click MATCH",
+      },
+    },
+    data: {
+      /** Row position of the stimulus */
+      stimulus_row: {
+        type: ParameterType.INT,
+      },
+      /** Column position of the stimulus */
+      stimulus_col: {
+        type: ParameterType.INT,
+      },
+      /** Whether this trial was a target */
+      is_target: {
+        type: ParameterType.BOOL,
+      },
+      /** Whether participant responded */
+      response: {
+        type: ParameterType.BOOL,
+      },
+      /** Response time in milliseconds */
+      response_time: {
+        type: ParameterType.INT,
+      },
+      /** Whether the response was correct */
+      correct: {
+        type: ParameterType.BOOL,
+      },
+    },
+  citations: '__CITATIONS__',
+};
+
+type Info = typeof info;
+
+/**
+ * **plugin-spatial-nback**
+ *
+ * Single trial spatial grid stimulus with response collection
+ *
+ * @author A. Hunter Farhat
+ * @version 1.0.0
+ * @see {@link https://github.com/farhat60/jspsych-contrib/packages/plugin-spatial-nback}
+ */
+class SpatialNbackPlugin implements JsPsychPlugin<Info> {
+  static info = info;
+
+  constructor(private jsPsych: JsPsych) {}
+
+  trial(display_element: HTMLElement, trial: TrialType<Info>) {
+
+    // Validate grid dimensions
+    if (trial.rows <= 1 && trial.cols <= 1) {
+      console.log('About to throw grid error'); // Add this line
+      throw new Error("Grid must have more than one cell. Both rows and cols cannot be 1 or less.");
+    }
+    
+    // Additional validations
+    if (trial.rows <= 0 || trial.cols <= 0) {
+      console.log('About to throw dimension error'); // Add this line
+      throw new Error("Grid dimensions must be positive integers. Rows and cols must be greater than 0.");
+    }
+    
+    if (trial.stimulus_row >= trial.rows || trial.stimulus_col >= trial.cols) {
+      console.log('About to throw position error'); // Add this line
+      throw new Error(`Stimulus position (${trial.stimulus_row}, ${trial.stimulus_col}) is outside grid bounds (${trial.rows}x${trial.cols}).`);
+    }
+
+    let trial_start_time: number;
+    let response_allowed = false;
+    let response_given = false;
+    let stimulus_timeout: number;
+    let isi_timeout: number;
+    let stimulus_hidden = false; // Track if stimulus has been hidden
+
+    // Generate random position if not specified
+    const stimulus_row = trial.stimulus_row ?? Math.floor(Math.random() * trial.rows);
+    const stimulus_col = trial.stimulus_col ?? Math.floor(Math.random() * trial.cols);
+
+    const createDisplay = (): void => {
+      
+      
+      let html = `
+        <div id="nback-container" style="
+          position: fixed;
```

Comment:
using `position: fixed` makes the plugin take over the DOM in an agressive way. Would be better if this HTML/cSS blended into the existing jspsych-content CSS. It does some auto centering and such already.